tests/elster_xml/grundsteuer/test_elster_data_representation.py

```diff
@@ -7,12 +7,256 @@
+
+        assert result.E7403151 == 2
+        assert result.E7403152 == 200
+
+    def test_flaechen_one_under_100_should_set_fields(self):
+        flaechen = [99, 100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 1
+        assert result.E7403152 == 100
+
+
+class TestEWeitereWohn:
+    def test_should_construct_from_dict_correctly(self):
+        weitere_wohnraeume = WeitereWohnraeumeDetails.parse_obj({"anzahl": 2, "flaeche": 42})
+
+        result = EWeitereWohn(weitere_wohnraeume)
+
+        assert result.E7403121 == 2
+        assert result.E7403122 == 42
+
+
+class TestEGaragen:
+    def test_should_construct_from_dict_correctly(self):
+        garagen_anzahl = GaragenAnzahl.parse_obj({"anzahl_garagen": 2})
+
+        result = EGaragen(garagen_anzahl)
+
+        assert result.E7403171 == 2
+
+
+class TestEAngDurchschn:
+    def test_wohnflaeche_under_60_should_set_others_to_none(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(59).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Wohn_Unter60 is not None
+        assert result.Wohn_60bis100 is None
+        assert result.Wohn_ab100 is None
+
+    def test_wohnflaeche_between_60_100_should_set_others_to_none(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(99).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Wohn_Unter60 is None
+        assert result.Wohn_60bis100 is not None
+        assert result.Wohn_ab100 is None
+
+    def test_wohnflaeche_from_100_should_set_others_to_none(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(100).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Wohn_Unter60 is None
+        assert result.Wohn_60bis100 is None
+        assert result.Wohn_ab100 is not None
+
+    def test_on_weitere_wohnraeume_flag_true_should_set_weitere_wohnraeume(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(10).with_weitere_wohnraeume(42, 2).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Weitere_Wohn is not None
+
+    def test_on_weitere_wohnraeume_flag_false_should_not_set_weitere_wohnraeume(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(10).with_weitere_wohnraeume(42, 2).parse()
+        gebaeude.weitere_wohnraeume.has_weitere_wohnraeume = False
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Weitere_Wohn is None
+
+
+class TestEAngWohn:
+    def test_on_ab_1949_should_contain_baujahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_baujahr("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403114 == "1959"
+
+    def test_on_vor_1949_should_not_contain_baujahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_baujahr("1959").parse()
+        gebaeude.ab1949.is_ab1949 = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403114 is None
```

Comment:
How about testing the vor1949 flag?

Reply:
Right, I actually missed this. Tests are updated.